visidata/sheets.py

```diff
@@ -1169,8 +1169,15 @@ def preloadHook(sheet):
 
 @VisiData.api
 def newSheet(vd, name, ncols, **kwargs):
-    return Sheet(name, columns=[SettableColumn(width=vd.options.default_width) for i in range(ncols)], **kwargs)
-
+    #use temporary column names until we have the sheet object that can assign names and keep track of them
+    rename_cols = [SettableColumn(width=vd.options.default_width, name=f'name{i}') for i in range(ncols)]
+    vs = Sheet(name, columns=rename_cols, **kwargs)
+    # For TableSheet, we cannot change colnames using setName() on vs.columns[i],
```

Comment:
This smells like unnecessary complexity. Ideally this would all work smoothly so that you wouldn't have to dig into it and have to explain why it's done in a funny way.  But I'm unmotivated to untangle it myself and I would be loath to inflict it on you, so we'll just take this as-is.  Thanks for your attention to the details!